### PR TITLE
Add introduction/constants/enumeration/uninitialized to CONVENTIONS.

### PR DESCRIPTION
**Pull request by fiveop**
I have added new sections to the file. I would like the usual suspects to read them critically and offer critique.

When have agreed to add certain versions I will add Issues to track our progress in getting the code to follow the conventions. For two of the sections they already exist in #254 and #264.

### CODE REVIEW

File: CONVENTIONS.md [hunk 1]
```diff
@@ -1,7 +1,38 @@
+In order to achieve our goal of wrapping [libc][libc] code in idiomatic rust
```
Comment (kamalmarhubi) [inline CONVENTIONS.md]: Really minor: add a `# Conventions` or similar title up here, and make the lower headings h2 (`##`)

**Comment by kamalmarhubi** [comment]
Oh I like this!

**Comment by kamalmarhubi** [comment]
I'll give @posborne and @arcnmx a chance to comment, but feel free to r=me if they take a while :-)

**Comment by posborne** [comment]
No objection to anything here.  Looks good -- we can add more over time.

**Comment by fiveop** [comment]
@homu r=@kamalmarhubi

**Pull request merged**